fix: Read keystore password from disk or stdin

dknopik opened the pull request:
## Issue Addressed

- #632

## Proposed Changes

Change `--password` to `--password-file` and make it optional. If it is provided, the password is read from the provided path, else it is read from stdin.

dknopik committed:
fix: Read keystore password from disk or stdin

dknopik committed:
fix typo

diegomrsantos commented on anchor/keysplit/src/lib.rs:
```diff
@@ -54,7 +55,11 @@ pub fn run_keysplitter(
-        .decrypt_keypair(shared.password.as_bytes())
+        .decrypt_keypair(
+            read_password(shared.password_file.as_deref())
+                .map_err(KeysplitError::Keystore)?
```
Should we do sth like https://github.com/sigp/anchor/blob/ff56d2c17b1935105755c44cc001ee4386bdaa81/anchor/keysplit/src/lib.rs#L51 ?

diegomrsantos commented on anchor/keysplit/src/lib.rs:
```diff
@@ -54,7 +55,11 @@ pub fn run_keysplitter(
-        .decrypt_keypair(shared.password.as_bytes())
+        .decrypt_keypair(
+            read_password(shared.password_file.as_deref())
+                .map_err(KeysplitError::Keystore)?
```
Should we add more context, like at line 52?

dknopik replied:
done!

diegomrsantos commented:
Can we add tests?

dknopik commented:
> Can we add tests?

what kind of tests do you have in mind? this being a function that involves interactiveness and is otherwise very simple find it hard to test meaningfully.

dknopik committed:
add context to error

diegomrsantos committed:
test: add comprehensive tests for password reading in keysplit

- Refactor read_password() to support dependency injection for testing
- Add read_password_from_file() and read_password_from_stdin() helpers
- Add process_password() to handle trimming and validation
- Add 17 unit tests covering file and stdin password reading
- Add tempfile dev-dependency for test fixtures
- Tests cover success cases, edge cases (empty, whitespace, CRLF), and unicode

diegomrsantos commented:
> > Can we add tests?
> 
> what kind of tests do you have in mind? this being a function that involves interactiveness and is otherwise very simple find it hard to test meaningfully.

Pushed a commit, let me know what you think

dknopik commented:
Sure. feels a bit much to me, but i am fine wiht it

diegomrsantos commented:
> Sure. feels a bit much to me, but i am fine wiht it

imo, using smaller functions makes it easier to understand
